modules/light-clients/07-tendermint/misbehaviour_test.go

```diff
@@ -172,14 +172,14 @@ func (suite *TendermintTestSuite) TestMisbehaviourValidateBasic() {
 			},
 			func(misbehaviour *ibctm.Misbehaviour) error {
 				// voteSet contains only altVal which is less than 2/3 of total power (height/1height)
-				wrongVoteSet := tmtypes.NewVoteSet(chainID, int64(misbehaviour.Header1.GetHeight().GetRevisionHeight()), 1, tmproto.PrecommitType, altValSet)
+				wrongVoteSet := tmtypes.NewExtendedVoteSet(chainID, int64(misbehaviour.Header1.GetHeight().GetRevisionHeight()), 1, tmproto.PrecommitType, altValSet)
```

Comment:
does this need to change `NewVoteSet` still [exists](https://github.com/cometbft/cometbft/blob/v0.38.0-rc2/types/vote_set.go#L80)

Reply:
going to guess no, and try reverting the change.

Reply:
Did this work when it was revert back to `NewVoteSet`? 

I see it was removed in [94b8071](https://github.com/cosmos/ibc-go/pull/3883/commits/94b8071cc5275feff073fd0e4cff10d74ea5e2d4) but now it has been changed back in [2952fdf](https://github.com/cosmos/ibc-go/pull/3883/commits/2952fdfb5ef5e418a655679c47e69be35412f821) ?

Reply:
I believe this is "works vs not-work"

Reply:
any idea why the former does not work anymore?

Reply:
Fairly certain extensions must be enabled for `MakeExtCommit` to function properly on the precommitment round. By using `NewVoteSet`, extensions are not enabled and the extension height isn't set

> [test panicked: problem with vote extension data when making extended commit of height 4; vote extension](suite.go:87: test panicked: problem with vote extension data when making extended commit of height 4; vote extension signature is missing; validator addr E410BC16AB81076F7BF458CFC82E809D913587B7, timestamp 2020-01-02 00:00:00 +0000 UTC)

I'm not sure what vote extensions are, but they seem to be required for abci++ changes

Reply:
During the precommitment round, ABCI will make a call to the callee (SDK) for `ExtendVote` which may return additional information to be used in the next execution round. See the [abci++ docs on extend vote](https://docs.cometbft.com/v0.37/spec/abci/abci++_basic_concepts.html#consensusblock-execution-methods)

We could disable the additional vote extension with the following:
```diff
index 59c9488b7..a74a94e3d 100644
--- a/modules/light-clients/07-tendermint/misbehaviour_test.go
+++ b/modules/light-clients/07-tendermint/misbehaviour_test.go
@@ -172,13 +172,13 @@ func (suite *TendermintTestSuite) TestMisbehaviourValidateBasic() {
                        },
                        func(misbehaviour *ibctm.Misbehaviour) error {
                                // voteSet contains only altVal which is less than 2/3 of total power (height/1height)
-                               wrongVoteSet := tmtypes.NewExtendedVoteSet(chainID, int64(misbehaviour.Header1.GetHeight().GetRevisionHeight()), 1, tmproto.PrecommitType, altValSet)
+                               wrongVoteSet := tmtypes.NewVoteSet(chainID, int64(misbehaviour.Header1.GetHeight().GetRevisionHeight()), 1, tmproto.PrecommitType, altValSet)
                                blockID, err := tmtypes.BlockIDFromProto(&misbehaviour.Header1.Commit.BlockID)
                                if err != nil {
                                        return err
                                }
 
-                               extCommit, err := tmtypes.MakeExtCommit(*blockID, int64(misbehaviour.Header2.GetHeight().GetRevisionHeight()), misbehaviour.Header1.Commit.Round, wrongVoteSet, altSignerArr, suite.now, true)
+                               extCommit, err := tmtypes.MakeExtCommit(*blockID, int64(misbehaviour.Header2.GetHeight().GetRevisionHeight()), misbehaviour.Header1.Commit.Round, wrongVoteSet, altSignerArr, suite.now, false)
                                misbehaviour.Header1.Commit = extCommit.ToCommit().ToProto()
                                return err

```

What do you think @damiannolan should we do tests with or without vote extension in 07-tendermint (we should use vote extension in the testing pkg)

Reply:
Hmm! That's a little confusing to me 😅 So from the [abci++ spec](https://github.com/cometbft/cometbft/blob/main/spec/abci/abci++_methods.md#ExtendVote) it says that `ExtendVote` is called on _prevote_ and `VerifyVoteExtension` is called on _precommit_. I guess this is what the docs are saying too, that `ExtendVote` allows the application to add arbitrary non-deterministic data **for** _precommit_ (i.e. before). 

After just reading a bit of that, I'm happy to do the tests with vote extensions if you are!

Reply:
I will revert back to no usage of extension. It isn't needed for this testing path. We mainly just need to get a valid commit